asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/RubyGemsPreloader.java

```diff
@@ -71,12 +72,49 @@ && isOptionWithValue(attributes, Attributes.SOURCE_HIGHLIGHTER, CODERAY)) {
         }
     }
 
+    public void preloadRequiredLibrariesCommandLine(RubyHash opts) {
```

Comment:
Yes, I also feel not happy about this atm.
There are a few differences between what the 2 methods do though.
The new implementation leans towards the behavior of the Asciidoctor Ruby CLI.
For example this sets the backend as an attribute instead of passing the attribute as an option:
https://github.com/asciidoctor/asciidoctor/blob/1e6357ec8006782c8f1c65abd76f797912065256/lib/asciidoctor/cli/options.rb#L51-L54

Reply:
I understand the change is to align with the Ruby CLI, but why not apply the same in all places?

Reply:
Sure, I can try to use one implementation that handles both options and attributes with symbols and strings so ngs like the maven plugin won’t break.

Reply:
Removed the duplicate implementation again.